pkg/nvml/device.go

```diff
@@ -2286,3 +2286,145 @@ func DeviceGetBusType(Device Device) (BusType, Return) {
 func (Device Device) GetBusType() (BusType, Return) {
 	return DeviceGetBusType(Device)
 }
+
+// nvml.DeviceSetDefaultFanSpeed_v2()
+func DeviceSetDefaultFanSpeed_v2(Device Device, Fan int) Return {
+	return nvmlDeviceSetDefaultFanSpeed_v2(Device, uint32(Fan))
+}
+
+func (Device Device) SetDefaultFanSpeed_v2(Fan int) Return {
+	return DeviceSetDefaultFanSpeed_v2(Device, Fan)
+}
+
+// nvml.DeviceGetMinMaxFanSpeed()
+func DeviceGetMinMaxFanSpeed(Device Device) (int, int, Return) {
+	var MinSpeed, MaxSpeed uint32
+	ret := nvmlDeviceGetMinMaxFanSpeed(Device, &MinSpeed, &MaxSpeed)
+	return int(MinSpeed), int(MaxSpeed), ret
+}
+
+func (Device Device) GetMinMaxFanSpeed() (int, int, Return) {
+	return DeviceGetMinMaxFanSpeed(Device)
+}
+
+// nvml.DeviceGetThermalSettings()
+func DeviceGetThermalSettings(Device Device, SensorIndex uint32) (GpuThermalSettings, Return) {
+	var PThermalSettings GpuThermalSettings
+	ret := nvmlDeviceGetThermalSettings(Device, SensorIndex, &PThermalSettings)
+	return PThermalSettings, ret
+}
+
+func (Device Device) GetThermalSettings(SensorIndex uint32) (GpuThermalSettings, Return) {
+	return DeviceGetThermalSettings(Device, SensorIndex)
+}
+
+// nvml.DeviceGetDefaultEccMode()
+func DeviceGetDefaultEccMode(Device Device) (EnableState, Return) {
+	var DefaultMode EnableState
+	ret := nvmlDeviceGetDefaultEccMode(Device, &DefaultMode)
+	return DefaultMode, ret
+}
+
+func (Device Device) GetDefaultEccMode() (EnableState, Return) {
+	return DeviceGetDefaultEccMode(Device)
+}
+
+// nvml.DeviceGetPcieSpeed()
+func DeviceGetPcieSpeed(Device Device) (int, Return) {
+	var PcieSpeed uint32
+	ret := nvmlDeviceGetPcieSpeed(Device, &PcieSpeed)
+	return int(PcieSpeed), ret
+}
+
+func (Device Device) GetPcieSpeed() (int, Return) {
+	return DeviceGetPcieSpeed(Device)
+}
+
+// nvml.DeviceGetGspFirmwareVersion()
+func DeviceGetGspFirmwareVersion(Device Device) (string, Return) {
+	Version := make([]byte, SYSTEM_DRIVER_VERSION_BUFFER_SIZE)
```

Comment:
SHould this be `NVML_GSP_FIRMWARE_VERSION_BUF_SIZE`?

Reply:
Weird. I swear I looked for a const named like this one before choosing this alternative one and couldn't find it. Thanks for pointing it out.

Reply:
Updated